controller/env/appenv.go

```diff
@@ -183,8 +175,124 @@ type AppHandler func(ae *AppEnv, rc *response.RequestContext)
 
 type AppMiddleware func(*AppEnv, http.Handler) http.Handler
 
-func NewAppEnv(c *edgeconfig.Config) *AppEnv {
+type authorizer struct {
+}
+
+func (a authorizer) Authorize(request *http.Request, principle interface{}) error {
+	//principle is an API Session
```

Comment:
i actually can't discern if this should be principal or principle but i feel like it's probably principal?

Reply:
I probable auto filled this out - so it is probably spelled that way in the code that defines the interface. Will check.

Reply:
Wasn't - fixed in multiple locations besides this one.